AEPIdentity/Sources/IdentityState.swift

```diff
@@ -283,6 +283,30 @@ class IdentityState {
         }
     }
 
+    /// Clears all identities and regenerates a new ECID value.
+    /// Saves identities to persistence and creates a new shared state and dispatches a new` resetComplete` event after operation completes.
+    /// - Parameters:
+    ///   - event: event which triggered the reset call
+    ///   - createSharedState: function which creates new shared states
+    ///   - eventDispatcher: function which dispatches a new `Event`
+    func resetIdentifiers(event: Event,
+                          createSharedState: ([String: Any], Event) -> Void,
+                          eventDispatcher: (Event) -> Void) {
+        // clear the properties
+        identityProperties = IdentityProperties()
```

Comment:
@emdobrin any thoughts on this?

Reply:
The opt-out hit is sent after an ECID is cleared, so saying it doesn't need to be sent because there are no more interactions with the old ECID doesn't seem valid to me. I think the point of the opt-out hit is to delink the old ECID from the ad id server-side. However, I didn't consider the case of any existing hits which may be queued, so sending the opt-out hit before the queued hits are processed would be incorrect.
As we are clearing all the identifiers, are any existing queued hits still valid? In the case of privacy opt-out we took the position that those hits were no longer needed as we were clearing the identifiers anyway. Should we consider this case the same, and clear the hits?

Reply:
@kevinlind I'd be open to also clearing the hits in this case.